### ISSUE TITLE
Suggestion: findFirst support

### ISSUE THREAD

**Issue by squirtle1990** [Zhuinden/realm-monarchy#20]
It would be swell if this library supported findFirstMappedWithChanges/findFirstCopiedWithChanges for those instances where you want to bind a single LiveData object from Realm to a view.

**Comment by Zhuinden** [comment]
I was entertaining the idea for a long time but technically applying a listener to the object obtained with `findFirst` is `RealmObjectChangeListener` which stops emitting if the object is deleted.

The reason why I did not add it is because even in our own Realm-based project I was using `findAll()`/`findAllAsync()` for reliably obtaining the first element with `findAll()` and `.first(null)`, even if the item gets deleted and recreated with the given ID.

`findFirst()` also wouldn't create a subscription, just like how currently findAllPaged doesn't either.

And `findFirstAsync()` is a ***mess***, I do not know any use-case where it is the behavior that a user would need.

So I'd just recommend using it as `LiveData<List<T>>` which is generally how I'd do it with Room too and how I did it with the SQLite wrapper we had before Room. Single object queries are tricky when the object doesn't exist, in which case you'd need a form of `LiveData<Optional<T>>`... and then, which Optional should I use? my own? :smile:
